addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_image.py

```diff
@@ -95,27 +95,24 @@ def __gather_extras(sockets_or_slots, export_settings):
     return None
 
 
-def __gather_mime_type(sockets_or_slots, export_settings):
+def __gather_mime_type(sockets_or_slots, export_image, export_settings):
     # force png if Alpha contained so we can export alpha
     for socket in sockets_or_slots:
         if socket.name == "Alpha":
             return "image/png"
 
-    if export_settings["gltf_image_format"] == "NAME":
-        extension = __get_extension_from_slot(sockets_or_slots, export_settings)
-        extension = extension.lower()
-        if extension in [".jpeg", ".jpg", ".png"]:
-            return {
-                ".jpeg": "image/jpeg",
-                ".jpg": "image/jpeg",
-                ".png": "image/png",
-            }[extension]
-        return "image/png"
+    if export_settings["gltf_image_format"] == "AUTO":
+        image = export_image.blender_image()
+        if image is not None:
+            if image.file_format == 'JPEG':
+                return "image/jpeg"
+            else:
```

Comment:
Maybe these nested `if` statements can be collapsed?

(Also, in other projects, ReSharper always complains if I put `else` after `return`.  The `else` can't do or mean anything after a `return` statement.  Feel free to take or leave this comment).

```
if image is not None and image.file_format == 'JPEG':
    return "image/jpeg"
return "image/png"
```

Reply:
Ok, I did that instead.